paper/forestecology.Rmd

```diff
@@ -0,0 +1,456 @@
+  mutate(foldID = spatial_block_scbi$foldID %>% factor())
+```
+
+Figure \ref{fig:scbi-spatial-information} illustrates the net effect of adding these two elements of information to the `growth_scbi` data frame. The location of each tree is marked with an integer indicating which fold it belongs to, where the folds are marked with solid lines. The color of each digit indicates whether the tree is part of the buffer region (and thus will only be considered as a competitor tree in our model) or is part of the interior of the study region (and thus is a focal tree whose growth is of modeled interest).
+
+```{r scbi-spatial-information, out.width="66%", fig.cap="Add spatial information: Buffer region and spatial cross-validation blocks (1 through 4). All trees in the interior of the study region (i.e. not part of buffer) will be the focal trees whose growth will be modeled."}
+ggplot() +
+  geom_sf(data = blocks_scbi, fill = "transparent", linetype = "dashed") +
+  geom_sf_text(data = growth_scbi %>% sample_n(1000), aes(label = foldID, col = buffer))
+```
+
+
+
+
+## Step 3: Identify all focal and corresponding competitor trees {#focal-vs-comp}
+
+The next step in our analysis sequence is to identify all focal trees and their corresponding competitor trees. The `create_focal_vs_comp()` functions performs these tasks and returns a new data frame of type `sf`. On top of the previous arguments `comp_dist` defining the competition neighborhood and `id` indicating which variable uniquely identifies each tree-stem, this function also requires an `sf` object representation of the spatial cross-validation blocks/folds; in our case, this was manually encoded in the `blocks_scbi` in Section \ref{spatial-information} while in our Appendix we present an example where this was performed using `spatialBlock()` from the `blockCV` package. We present the resulting data frame below with the `foldID` variable omitted for compactness of presentation.
+
+```{r scbi-focal-vs-comp}
+# TODO: Below reconcile the number of rows as they off by one from growth_scbi %>% 
+# filter(!is.na(growth) & !buffer). Perhaps by removing NA's in the growth_scbi stage?
+focal_vs_comp_scbi <- growth_scbi %>%
+  create_focal_vs_comp(comp_dist, cv_grid_sf = blocks_scbi, id = "stemID")
+focal_vs_comp_scbi %>% 
+  select(-foldID)
+```
+
+The resulting data frame `focal_vs_comp_scbi` has `r nrow(focal_vs_comp_scbi)` rows, representing the subset of the `r nrow(growth_scbi)` trees in `growth_scbi` that will be considered as focal trees. Two new variables `focal_ID` and `focal_sp` relate to tree-stem identification and species information. Most notably however is a new variable `comp` which contains information on all competitor trees for a given focal tree, saved in `tidyr` package list-column format @tidyr_package. For example, we drill-down on the tree with `focal_ID` 4, which has 20 competitor trees each described by 4 variables as indicated by the fact that `comp` is a `<tibble [20 × 4]>`. 
```

Comment:
I think `<tibble [20 × 4]>` would be the print format if we had called `str()` on the tibble? If we're just printing, should we do `tibble: 20 x 4`?